flink-streaming-java/src/test/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironmentComplexConfigurationTest.java

```diff
@@ -222,7 +223,7 @@ public void testGettingEnvironmentWithConfiguration() {
 
         assertThat(env.getParallelism(), equalTo(10));
         assertThat(env.getConfig().getAutoWatermarkInterval(), equalTo(100L));
-        assertThat(env.getStateBackend(), instanceOf(MemoryStateBackend.class));
+        assertThat(env.getStateBackend(), nullValue());
```

Comment:
I would remove this verification and the setting of state.backend.

Reply:
By the way, it's better to migrate the test to junit5 if changing a junt4 test.